crates/cubecl-runtime/src/server.rs

```diff
@@ -149,6 +148,90 @@ impl Handle {
     }
 }
 
+/// Bindings to execute a kernel.
+#[derive(Debug, Default)]
+pub struct Bindings {
+    /// Buffer bindings
+    pub buffers: Vec<Binding>,
+    /// Packed metadata for tensor bindings (len, shape, stride, etc).
+    /// Ordered by inputs, then outputs, then tensormaps
+    pub metadata: MetadataBinding,
+    /// Scalar bindings
+    pub scalars: Vec<ScalarBinding>,
+    /// Tensor map bindings
+    pub tensor_maps: Vec<TensorMapBinding>,
+}
+
+impl Bindings {
+    /// Create a new bindings struct
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Add a buffer binding
+    pub fn with_buffer(mut self, binding: Binding) -> Self {
+        self.buffers.push(binding);
+        self
+    }
+
+    /// Set the buffers to `bindings`
+    pub fn with_buffers(mut self, bindings: Vec<Binding>) -> Self {
+        self.buffers = bindings;
+        self
```

Comment:
That's probably dangerous, I would expect with_buffers == with_buffer multiple times, but it's not since it removes all previously added values.

Reply:
Yeah I wasn't sure, I guess the field is public so you could just set it directly if you intend to override it. I'll change it.